Code/Tools/AssetProcessor/native/ui/ProductDependencyTreeItemData.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <native/ui/ProductDependencyTreeItemData.h>
+
+#include <AzCore/std/smart_ptr/make_shared.h>
+
+#include <QDir>
+#include <QStack>
+
+namespace AssetProcessor
+{
+    AZStd::shared_ptr<ProductDependencyTreeItemData> ProductDependencyTreeItemData::MakeShared(
+        QString name, AZStd::string productName)
+    {
+        return AZStd::make_shared<ProductDependencyTreeItemData>(name, productName);
+    }
+
+    ProductDependencyTreeItemData::ProductDependencyTreeItemData(
+        QString name, AZStd::string productName)
+        : m_name(name)
+        , m_productName(productName)
+    {
+    }
+
+    ProductDependencyTreeItem::ProductDependencyTreeItem(
+        AZStd::shared_ptr<ProductDependencyTreeItemData> data, ProductDependencyTreeItem* parentItem)
+        : m_data(data)
+        , m_parent(parentItem)
+    {
+    }
+
+    ProductDependencyTreeItem::~ProductDependencyTreeItem()
+    {
+    }
+
+    ProductDependencyTreeItem* ProductDependencyTreeItem::CreateChild(AZStd::shared_ptr<ProductDependencyTreeItemData> data)
+    {
+        m_childItems.emplace_back(new ProductDependencyTreeItem(data, this));
+        return m_childItems.back().get();
```

Comment:
I think emplace_back returns the object you just inserted return `m_childItems.emplace_back(new ProductDependencyTreeItem(data, this)).get()`